src/com/amazon/ion/impl/IonCursorBinary.java

```diff
@@ -178,8 +262,362 @@ class IonCursorBinary implements IonCursor {
         this.offset = offset;
         this.limit = offset + length;
         byteBuffer = ByteBuffer.wrap(buffer, offset, length);
+        refillableState = null;
+    }
+
+    /**
+     * The standard {@link IonBufferConfiguration}. This will be used unless the user chooses custom settings.
+     */
+    private static final IonBufferConfiguration STANDARD_BUFFER_CONFIGURATION =
+        IonBufferConfiguration.Builder.standard().build();
+
+    /**
+     * @param value a non-negative number.
+     * @return the exponent of the next power of two greater than or equal to the given number.
+     */
+    private static int logBase2(int value) {
+        return 32 - Integer.numberOfLeadingZeros(value == 0 ? 0 : value - 1);
+    }
+
+    /**
+     * @param value a non-negative number
+     * @return the next power of two greater than or equal to the given number.
+     */
+    static int nextPowerOfTwo(int value) {
+        return (int) Math.pow(2, logBase2(value));
+    }
```

Comment:
```suggestion
    static int nextPowerOfTwo(int value) {
        long highBit = Integer.toUnsignedLong(Integer.highestOneBit(value));
        // Paraphrased from JLS 5.1.3. Narrowing Primitive Conversion
        // For a narrowing conversion of a floating-point number to an integral type T
        // If the input is not NaN and is not representable as the target type int, then the value must be too large
        // (a positive value of large magnitude or positive infinity), and the result is the largest representable
        // value of type int.
        return (int)(double) ((highBit == value) ? value : highBit << 1);
    }
```

I tried a bit of JMH benchmarking on the two options:
```
Benchmark                     (arg)  Mode  Cnt   Score   Error  Units
MyBenchmark.original              1  avgt   15  15.908 ± 6.022  ns/op
MyBenchmark.original              2  avgt   15  17.654 ± 0.929  ns/op
MyBenchmark.original             35  avgt   15  18.145 ± 1.349  ns/op
MyBenchmark.original     2147483647  avgt   15  22.419 ± 3.588  ns/op
MyBenchmark.original    -2147483648  avgt   15  20.333 ± 0.765  ns/op
MyBenchmark.suggestion            1  avgt   15   1.479 ± 0.051  ns/op
MyBenchmark.suggestion            2  avgt   15   1.459 ± 0.029  ns/op
MyBenchmark.suggestion           35  avgt   15   1.466 ± 0.026  ns/op
MyBenchmark.suggestion   2147483647  avgt   15   4.206 ± 0.186  ns/op
MyBenchmark.suggestion  -2147483648  avgt   15   4.143 ± 0.063  ns/op
```

Reply:
The insight to me is. that `Math.pow` is not needed (and is the likeliest perf. cost).

To that end, I stumbled on the below in the ORC writer implementation today:
```
    final int pow2 = size == 1 ? 1 : Integer.highestOneBit(size - 1) * 2;
```

Reply:
The ORC writer implementation (alone, as written there) has additional failure modes and different behavior than `nextPowerOfTwo`- it will wrap (go negative) and not return the same value when the input already *is* a power of two.

Reply:
This is not on the critical performance path (growth happens very infrequently), so I'm inclined to go with whatever we find the most readable while retaining the same behavior. I'm happy to go with Josh's suggestion if we think that's more readable than the existing proposal.

Reply:
> and not return the same value when the input already is a power of two.
FTR it does. The reason I shared is that's the genius of subtracting 1 from the value.

Anyway, I think Josh's proposal is fine. I think either strategy that finds the highest 1 bit is clear.

Reply:
Accepted Josh's recommendation